HW0/P5/P5.txt

```diff
@@ -1 +1,15 @@
 # Your answers here.
+
+1. If one cashier were to add the totals of 256 bags, he would have to calculate 255 additions. This is because he can add separate pairs of the 256 numbers, taking 128 seconds. Subsequently, he can add separate pairs of the remaining 128 numbers, taking 64 seconds. He then adds separate pairs of the 64 numbers, taking 32 seconds...so on and so forth. Hence, he takes a total of 128+64+32+16+8+4+2+1 = 255 seconds.
+
+2. Together, the 7 cashier co-workers and the original cashier can verify the total sum faster. In the first round of counting, each one of the 8 cashiers can be separately in charge of 256/8=32 numbers, each reducing them to 16 numbers and taking 16 seconds in total. Subsequently, each cashier reduces their 16 numbers to 8 numbers, taking 8 seconds...so on and so forth. After 16+8+4+2+1 = 31 seconds, there will be 8 numbers left, which takes 3 more seconds. Hence, 34 seconds needed in total.
+
+3. The employees can verify the sum in 8 seconds. This is because the infinite number of employees would take 1 minute to half the number of bags to 128, another 1 minute to half those number of bags to 64, so on. Since 256 is 2 to the power of 8, it would take 8 seconds total.
+
+4. If there were N bags, it would take log_2(N) seconds for the infinite number of cashiers to count the bags. The graph is saved as P5.png in this directory.
```

Comment:
In your code you have "y1 = [math.log(i) for i in x]", think this uses base e not 2